applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py

```diff
@@ -0,0 +1,156 @@
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, model, custom_settings):
+        # Set defaults and validate custom settings.
+        adaptative_remesh_parameters = KratosMultiphysics.Parameters("""
+        {
+            "compute_error_settings" : {
+                "error_mesh_tolerance" : 5.0e-3,
+                "error_mesh_constant"  : 5.0e-3,
+                "compute_error_extra_parameters":
+                {
+                    "stress_vector_variable"              : "CAUCHY_STRESS_VECTOR",
+                    "echo_level"                          : 0
+                }
+            },
+            "metric_error_parameters" :
+            {
+                "minimal_size"                        : 0.01,
+                "maximal_size"                        : 1.0,
+                "error_strategy_parameters":
+                {
+                    "target_error"                        : 0.01,
+                    "set_target_number_of_elements"       : false,
+                    "target_number_of_elements"           : 1000,
+                    "perform_nodal_h_averaging"           : false
+                },
+                "echo_level"                          : 0
+            },
+            "remeshing_parameters":
+            {
+                "filename"                             : "out",
+                "framework"                            : "Lagrangian",
+                "internal_variables_parameters"        :
+                {
+                    "allocation_size"                      : 1000,
+                    "bucket_size"                          : 4,
+                    "search_factor"                        : 2,
+                    "interpolation_type"                   : "LST",
+                    "internal_variable_interpolation_list" :[]
+                },
+                "save_external_files"              : false,
+                "max_number_of_searchs"            : 1000,
+                "echo_level"                       : 0
+            }
+        }
+        """)
+
+        # Validate the remaining settings in the base class.
+        self.validate_and_transfer_matching_settings(custom_settings, adaptative_remesh_parameters)
+        self.adaptative_remesh_parameters = adaptative_remesh_parameters
+
+        # Construct the base solver.
+        super(AdaptativeStaticMechanicalSolver, self).__init__(model, custom_settings)
+        self.print_on_rank_zero("::[AdaptativeStaticMechanicalSolver]:: ", "Construction finished")
+
+    #### Private functions ####
+
+    def AddVariables(self):
+        super(AdaptativeStaticMechanicalSolver, self).AddVariables()
+        if (missing_meshing_dependencies is False):
+            self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_H)
+        self.print_on_rank_zero("::[AdaptativeStaticMechanicalSolver]:: ", "Variables ADDED")
+
+    def get_remeshing_process(self):
```

Comment:
This has been discussed before but just to add my thoughts:

The processes live on the AnalysisStage and should not be in the solver.
Question: Does the remeshing-process have to be a process or could it be a utility and thus avoiding all this naming struggle?

Reply:
This is the only way to reduce code duplication, because otherwise you need to copy this in each solver